### PR TITLE
LibGfx: Stop assuming the target format is always BGRA8888

### LINKED ISSUE #18749: LibGfx: Color blending issue (probably alpha channel related)
I just noticed this odd looking color blending issue that I've only seen with the 3D File Viewer. WindowServer uses `Painter::fill_rect` with the rubber band fill color, which defaults to `#f4ca9e3c`.

[color_blend_issue.webm](https://github.com/SerenityOS/serenity/assets/10320822/a9a9ddb6-1d16-4f47-80b2-3636e678b743)

### PR DESCRIPTION
**Pull request by MacDue**
...and instead assume it's BGRx8888 or BGRA8888, for now. Always treating the target as BGRA8888 leads to the alpha channel being interpreted incorrectly sometimes (as can be seen with WindowServer overlays).

Fixes #18749

### CODE REVIEW

File: Userland/Libraries/LibGfx/Painter.cpp [hunk 1]
```diff
@@ -121,9 +134,10 @@ void Painter::fill_physical_rect(IntRect const& physical_rect, Color color)
     ARGB32* dst = m_target->scanline(physical_rect.top()) + physical_rect.left();
     size_t const dst_skip = m_target->pitch() / sizeof(ARGB32);
 
+    auto dst_format = target()->format();
     for (int i = physical_rect.height() - 1; i >= 0; --i) {
         for (int j = 0; j < physical_rect.width(); ++j)
-            dst[j] = Color::from_argb(dst[j]).blend(color).value();
+            dst[j] = color_for_format(dst_format, dst[j]).blend(color).value();
```
Comment (awesomekling) [inline Userland/Libraries/LibGfx/Painter.cpp:140]: It seems unfortunate to add a branch on a runtime-constant value in such a hot function.
Is CPU branch prediction good enough that this is not a problem? Or could we make it faster by duplicating the loop (e.g via C++ template) for each supported format?

Comment (MacDue) [reply Userland/Libraries/LibGfx/Painter.cpp:140]: It seems _mostly_ pretty negligible, at least in `blit_filtered()`, I see a (1-2%) regression with this branch (when all formats are BGRA8888, so always use `from_argb()`). 

That said, there is a order of magnitude slowdown between `Color::from_argb()` and `Color::from_rgb()`. It seems that `| 0xff000000` is surprisingly costly :^(, and this slowdown is observed whether templates or not.

My benchmark takes 2.6 seconds when everything is BGRA and 71 seconds if using BGRx.

Comment (MacDue) [reply Userland/Libraries/LibGfx/Painter.cpp:140]: I'll give it a try and see...

Comment (MacDue) [reply Userland/Libraries/LibGfx/Painter.cpp:140]: Doing that seems to be about 11687ms (so the largest regression, seems a branch is preferred over function pointer indirection).

Comment (konradekk) [reply Userland/Libraries/LibGfx/Painter.cpp:140]: thanks! 🙇🏻‍♂️

**Pull request merged**